tests/temp_mixin.py

```diff
@@ -9,14 +9,14 @@ class TempMixin:
     """
 
     def setUp(self):
-        super().setUp()
         self.tempdir = tempfile.mkdtemp()
+        super().setUp()
```

Comment:
If `super().setUp()` fails here, the temporary directory is created but then the `tearDown` would not be called.  Instead, have a `remove_tempdir` method and `self.addCleanup(self.remove_tempdir)` right after `mkdtemp` would ensure that the temporary directory is removed.